lib/auth/auth.go

```diff
@@ -3880,11 +3880,11 @@ func (a *Server) deleteMFADeviceSafely(ctx context.Context, user, deviceName str
 	// Check whether the device to delete is the last passwordless device,
 	// and whether deleting it would lockout the user from login.
 	//
-	// TODO(Joerger): the user may already be locked out from login if a password
+	// Note: the user may already be locked out from login if a password
```

Comment:
As you said, it's a pretty weird edge case, but I think I'm OK with the if essentially saying "is last passkey && has no password".